src/Sema.zig

```diff
@@ -9966,6 +9975,22 @@ fn finishFunc(
         });
     }
 
+    if (cc_resolved == .Interrupt and return_type.zigTypeTag(mod) != .Void) {
+        const msg = msg: {
+            const msg = try sema.errMsg(
+                block,
+                ret_ty_src,
+                "non-void return type '{}' not allowed in function with calling convention 'Interrupt'",
+                .{return_type.fmt(mod)},
+            );
+            errdefer msg.destroy(gpa);
+
+            try sema.addDeclaredHereNote(msg, return_type);
+            break :msg msg;
+        };
+        return sema.failWithOwnedErrorMsg(block, msg);
+    }
```

Comment:
`fail` adds `addDeclaredHereNote` automatically.
```suggestion
    if (cc_resolved == .Interrupt and return_type.zigTypeTag(mod) != .Void) {
        return sema.fail(
            block,
            cc_src,
            "non-void return type '{}' not allowed in function with calling convention 'Interrupt'",
            .{return_type.fmt(mod)},
        );
    }
```